sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -238,71 +238,77 @@ def build_task_spec_for_task(
                     input_name].component_input_parameter = (
                         component_input_parameter)
 
-        elif isinstance(input_value, str):
-            # Handle extra input due to string concat
-            pipeline_channels = (
-                pipeline_channel.extract_pipeline_channels_from_any(input_value)
-            )
-            for channel in pipeline_channels:
-                # value contains PipelineChannel placeholders which needs to be
-                # replaced. And the input needs to be added to the task spec.
-
-                # Form the name for the compiler injected input, and make sure it
-                # doesn't collide with any existing input names.
-                additional_input_name = (
-                    compiler_utils.additional_input_name_for_pipeline_channel(
-                        channel))
-
-                # We don't expect collision to happen because we prefix the name
-                # of additional input with 'pipelinechannel--'. But just in case
-                # collision did happend, throw a RuntimeError so that we don't
-                # get surprise at runtime.
-                for existing_input_name, _ in task.inputs.items():
-                    if existing_input_name == additional_input_name:
-                        raise RuntimeError(
-                            f'Name collision between existing input name {existing_input_name} and compiler injected input name {additional_input_name}'
-                        )
-
-                additional_input_placeholder = placeholders.InputValuePlaceholder(
-                    additional_input_name)._to_string()
-                input_value = input_value.replace(channel.pattern,
-                                                  additional_input_placeholder)
-
-                if channel.task_name:
-                    # Value is produced by an upstream task.
-                    if channel.task_name in tasks_in_current_dag:
-                        # Dependent task within the same DAG.
-                        pipeline_task_spec.inputs.parameters[
-                            additional_input_name].task_output_parameter.producer_task = (
-                                utils.sanitize_task_name(channel.task_name))
-                        pipeline_task_spec.inputs.parameters[
-                            additional_input_name].task_output_parameter.output_parameter_key = (
-                                channel.name)
+        elif isinstance(input_value, (str, int, float, bool, dict, list)):
+            if isinstance(input_value, (str, dict, list)):
```

Comment:
you can remove this `if isinstance(input_value, (str, dict, list)):` check, `extract_pipeline_channels_from_any` would return an empty list in case float, int, bool.

Reply:
Is it fine to remove the inner if check and expand the type annotations?

It would simplify the code but I'm not sure if it makes sense to update the type annotations for `extract_pipeline_channels_from_any` since ints, floats, and bools can't contain pipeline channels.
 https://github.com/kubeflow/pipelines/blob/92c317824a3697d955cb78cc9a85ecac58a3366c/sdk/python/kfp/dsl/pipeline_channel.py#L589

Reply:
Yes, you can update the payload annotation type.

Reply:
Done.